### PR TITLE
Added a method that allows the generation of report without writing to a file

### PR DESCRIPTION
**Pull request by nicwillemse**
Added a method that allows the generation of report without writing to a file

### CODE REVIEW

**Comment by nicwillemse** [comment]
@cmckni3 any idea what the problem might be here ? This does not seem to be related to the code I have added ?

**Comment by cmckni3** [comment]
@nicwillemse Strange. This is actually a bundler error.

File: lib/jasperserver-rails/jasperserver-dsl.rb [hunk 1]
```diff
@@ -37,11 +37,12 @@ def run_report(filename, &block)
         ).to_s,
         { cookies: @cookie }
       )
+    end
 
-      # Write file
+    def run_report(filename, &block)
```
Comment (cmckni3) [inline lib/jasperserver-rails/jasperserver-dsl.rb]: Excellent. This is the reason I reverted ffe38af4771ddefee27cb23062b3388e1a03474e. I wanted a separate method.

**Comment by nicwillemse** [comment]
@cmckni3 do you have any capacity to look at this now ?

**Comment by cmckni3** [comment]
@nicwillemse Thanks a lot for this contribution. I really meant to get around to making this change after I reverted ffe38af4771ddefee27cb23062b3388e1a03474e due to the control coupling.

The coupling made the method more complicated in my opinion and made testing a little awkward.

**Comment by cmckni3** [comment]
@nicwillemse I am doing a little investigation. If I can't figure it out in 15 minutes, I will have to wait about 8 hours.

**Comment by nicwillemse** [comment]
only a pleasure thanks @cmckni3 !

**Comment by cmckni3** [comment]
Running a new travis build on the container infrastructure to see if it fixes the issue.

**Comment by cmckni3** [comment]
@nicwillemse Can you rebase your changes on top of master? The build should pass on Travis after triggering a new build.

**Comment by nicwillemse** [comment]
@cmckni3 - I have done a merge, I realise a rebase would have been better but there it is

**Comment by cmckni3** [comment]
:+1:

**Comment by cmckni3** [comment]
Will create a new Rubygems release in the morning

**Comment by cmckni3** [comment]
Fixes #5

**Comment by nicwillemse** [comment]
Awesome - much appreciated @cmckni3

**Comment by cmckni3** [comment]
Published to rubygems

**Pull request merged**